### ISSUE TITLE
assertReverseUrlMapping fails for url mapping with defined plugin

### ISSUE THREAD

**Issue by thojo** [grails/grails-core#9863]
Using the URL mapping (with `grails.web.url.converter = 'hyphenated'`) ...

``` groovy
class UrlMappings {
    static mappings = {
        '/my-url'(controller: 'myController', plugin: 'myPlugin')
    }
}
```

... the test ...

``` groovy
@TestFor(UrlMappings)
@Mock(MyController)
class UrlMappingsSpec extends Specification {
    void 'my test'() {
        expect:
        assertReverseUrlMapping('/my-url', controller: 'myController', plugin: 'myPlugin')
    }
}
```

... fails with `reverse mapping assertion for {controller = myController, action = null, params = [:]} expected:<[/my-url]> but was:<[]>`.

Removing `plugin: 'myPlugin'` from the URL mapping the test succeeds but I have to use a Controller from a specific plugin.

**Comment by thojo** [comment]
A fix for the 2.5 branch would be great! I can't yet use Grails 3.

**Comment by snimavat** [comment]
This issue could be reproduced with 3.x as well, just gave a pull request with tests..
